.github/workflows/create-block.yml

```diff
@@ -26,8 +26,8 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                node: ['20', '22', '24']
-                os: ['macos-15', 'ubuntu-24.04', 'windows-2025']
+                node: ${{ github.event_name == 'pull_request' && fromJSON('["20"]') || fromJSON('["20", "22", "24"]') }}
```

Comment:
Instead of this, which is pretty messy and hard to read, what if there was an `event` key in the matrix that contains `github.event_name`. Then we can just have `exclude` statements for  `macos-15` & `pull_request`, `windows-2025` & `pull_request`, and then any additional versions that we'd like to exclude.